### ISSUE TITLE
Accessibility: Several sub-components in environment manager lacks accessibility

### ISSUE THREAD

**Issue by anthonykim1** [microsoft/vscode-python-environments#61]
Testing #24503

Couple components in the environment manager seem to have room for improvement in terms of accessibility.

1. "Create Python Terminal" case:
On screen reader mode, clicking on create Python terminal on environment manager doesnt give description/context of what is happening but rather just reads what goes inside terminal. 
I assume this may be very confusing to screen reader user because of sudden change of focus to terminal without any notification/description. 

https://github.com/user-attachments/assets/1894b2a6-91d8-4e5a-ad79-f6efbba83e8b

The check mark icon which is right next to the terminal icon does a better job in a way that it tells "browse" so that the screen reader user should have better context. Not too sure why it wont say "terminal" or "create python terminal" for the terminal icon. 

2. Create environment icon and refresh environment icon case:
Similarly, the create environment icon and refresh environment icon also does not provide any voice context over when pressed in screen reader mode (with voice over on)
 ![Image](https://github.com/user-attachments/assets/a172ded9-3928-4ffe-bcfa-cbd21e4857e2)

**Comment by karthiknadig** [comment]
@meganrogge What can we do here to make it better?

**Comment by meganrogge** [comment]
Re scenario number 1, does focusing the terminal button _with the keyboard_ provide an informative aria label, like "create python terminal"? If not, I recommend adding an aria label to the button to indicate what will happen. The resulting terminal focus info is already enough to indicate that has happened. 

![Image](https://github.com/user-attachments/assets/edbbd36b-052e-4ddc-ab65-ec18332fb10f)


Re scenario number 2, for refreshing, could we use a progress indicator? 
```
export interface Progress<T> {

		/**
		 * Report a progress update.
		 * @param value A progress item, like a message and/or an
		 * report on how much work finished
		 */
		report(value: T): void;
	}
```

For creating the environment, what gets focused when that is pressed?

**Comment by anthonykim1** [comment]
> For creating the environment, what gets focused when that is pressed? 

@meganrogge Right now, that would open the quick pick like:  

<img width="921" height="645" alt="Image" src="https://github.com/user-attachments/assets/1f89cc6c-c590-4fed-8947-1dc68f36011a" />

PR for progress indicator: https://github.com/microsoft/vscode-python-environments/pull/1205
